dev/io.openliberty.security.jakartasec.4.0.internal/src/io/openliberty/security/jakartasec40/identitystore/InMemoryIdentityStore.java

```diff
@@ -0,0 +1,156 @@
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package io.openliberty.security.jakartasec40.identitystore;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+
+import com.ibm.websphere.ras.ProtectedString;
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.ras.annotation.Sensitive;
+
+import io.openliberty.security.jakartasec40.identitystore.InMemoryIdentityStoreDefinitionWrapper.CredentialValue;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Default;
+import jakarta.security.enterprise.CallerPrincipal;
+import jakarta.security.enterprise.credential.CallerOnlyCredential;
+import jakarta.security.enterprise.credential.Credential;
+import jakarta.security.enterprise.credential.UsernamePasswordCredential;
+import jakarta.security.enterprise.identitystore.CredentialValidationResult;
+import jakarta.security.enterprise.identitystore.IdentityStore;
+import jakarta.security.enterprise.identitystore.InMemoryIdentityStoreDefinition;
+
+/**
+ * Liberty's InMemory {@link IdentityStore} implementation.
+ */
+@Default
+@ApplicationScoped
+@Component(service = {},
+           immediate = true,
+           configurationPolicy = ConfigurationPolicy.IGNORE,
+           property = "service.vendor=IBM")
+public class InMemoryIdentityStore implements IdentityStore {
+
+    private static final TraceComponent tc = Tr.register(InMemoryIdentityStore.class);
+
+    /** The definitions for this IdentityStore. */
+    private final InMemoryIdentityStoreDefinitionWrapper inMemoryIdentityStoreDefinitionWrapper;
+
+    /**
+     * Construct a new {@link InMemoryIdentityStore} instance using the specified definitions
+     * from the annotation.
+     *
+     * @param inMemoryIdentityStoreDefinition The definitions to use to configure the {@link IdentityStore}.
+     */
+
+    public InMemoryIdentityStore(InMemoryIdentityStoreDefinition inMemoryIdentityStoreDefinition) {
+        inMemoryIdentityStoreDefinitionWrapper = new InMemoryIdentityStoreDefinitionWrapper(inMemoryIdentityStoreDefinition);
+    }
+
+    @Override
+    public Set<String> getCallerGroups(CredentialValidationResult validationResult) {
+        if ((validationResult == null) || (!validationTypes().contains(ValidationType.PROVIDE_GROUPS))) {
+            return Collections.emptySet();
+        }
+
+        CallerPrincipal callerPrincipal = validationResult.getCallerPrincipal();
+        String caller = (callerPrincipal != null) ? callerPrincipal.getName() : null;
+
+        if (caller == null) {
+            if (tc.isDebugEnabled()) {
+                Tr.event(tc, "A null caller was passed into getCallerGroups. No groups returned." + validationResult);
```

Comment:
```suggestion
                Tr.debug(tc, "A null caller was passed into getCallerGroups. No groups returned." + validationResult);
```
You changed the `tc.isEventEnabled`, but you didn't change `Tr.event`

Reply:
Oh crimey, I do appologize!